swarm/api/http/server.go

```diff
@@ -841,15 +838,24 @@ func (s *Server) HandleGetFile(w http.ResponseWriter, r *Request) {
 			Respond(w, r, fmt.Sprintf("cannot resolve %s: %s", r.uri.Addr, err), http.StatusNotFound)
 			return
 		}
-		resolvedKey = true // url was of type bzz://name.eth/path , so we won't set cache headers
 	} else {
-		resolvedKey = false // url was of type bzz://<hex key>/path, so we are sure it is immutable.
+		w.Header().Set("Cache-Control", "max-age=2147483648, immutable") // url was of type bzz://<hex key>/path, so we are sure it is immutable.
 	}
 
 	log.Debug("handle.get.file: resolved", "ruid", r.ruid, "key", manifestKey)
 
 	reader, contentType, status, contentKey, err := s.api.Get(manifestKey, r.uri.Path)
 
+	etag := common.Bytes2Hex(contentKey)
+	noneMatchEtag := r.Header.Get("If-None-Match")
+	w.Header().Set("ETag", etag) // set etag to actual content key.
+	if noneMatchEtag != "" {
+		if bytes.Equal(storage.Key(common.Hex2Bytes(noneMatchEtag)), contentKey) {
+			Respond(w, r, "Not Modified", 304)
```

Comment:
please replace `304` with `http.StatusNotModified`

Reply:
could you push? can't see it

Reply:
My apologies. Pushed now. Check gitter also if you can. Thanks!